src/apex/queuing.rs

```diff
@@ -77,48 +77,118 @@ pub mod abstraction {
     use crate::apex::types::basic::PortDirection;
     use crate::prelude::*;
 
-    #[derive(Debug)]
-    pub struct QueuingPortSender<
+    #[derive(Debug, Clone)]
+    pub struct ConstQueuingPortSender<
         const MSG_SIZE: MessageSize,
         const NB_MSGS: MessageRange,
         Q: ApexQueuingPortP4Ext,
-    > {
+    >(QueuingPortSender<Q>);
+
+    impl<const MSG_SIZE: MessageSize, const NB_MSGS: MessageRange, S: ApexQueuingPortP4Ext>
+        AsRef<QueuingPortSender<S>> for ConstQueuingPortSender<MSG_SIZE, NB_MSGS, S>
+    {
+        fn as_ref(&self) -> &QueuingPortSender<S> {
+            &self.0
+        }
+    }
```

Comment:
Should this type also implement `Deref`? I'd like to be able to use the `Const*` version everywhere where the non-const version is required without calling `as_ref`.

Reply:
Looks like I made a mistake. I meant to implement `Deref` instead

Reply:
You generally should implement both if you implement Deref.

Reply:
https://doc.rust-lang.org/std/convert/trait.AsRef.html#generic-implementations